src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -2286,7 +2286,8 @@ private string GetLocaleInfoCoreUserOverride(LocaleStringData type)
             if (GlobalizationMode.Invariant)
                 return null!;
 
-            return ShouldUseUserOverrideNlsData ? NlsGetLocaleInfo(type) : IcuGetLocaleInfo(type);
+            // For ListSeparator on Windows, we don't call ICU as we fallback there to the thousands separator and getting the value from NLS would better here.
+            return ShouldUseUserOverrideNlsData || (type == LocaleStringData.ListSeparator && IsWin32Installed) ? NlsGetLocaleInfo(type) : IcuGetLocaleInfo(type);
```

Comment:
One of the goals for defaulting to use ICU was behavioral conformance across OSes; falling back to NLS here to fill in some of the data would seem to go against that goal.  Is this something to be concerned about?

Reply:
The returned value when we are using ICU is not perfect and we are looking at improving this in 6.0. Having inconsistency between Windows and Linux is better than the results we are getting on Windows. I am having this change for now here as it is candidate to port to 5.0. But for 6.0 release we should have cleaner solution which include consistency.